src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -4,57 +4,108 @@ package freechips.rocketchip.diplomacy
 {
+  /** Contains sub-[[LazyModule]]s; can be accessed by [[getChildren]]. */
   protected[diplomacy] var children = List[LazyModule]()
+  /** Contains the [[BaseNode]]s instantiated within this instance. */
   protected[diplomacy] var nodes = List[BaseNode]()
+  /** Stores [[SourceInfo]] of this instance.
+    *
+    * The companion object factory method will set this to the correct value.
+    */
   protected[diplomacy] var info: SourceInfo = UnlocatableSourceInfo
-  protected[diplomacy] val parent = LazyModule.scope
+  /** Parent of this LazyModule. If this instance is at the top of the hierarchy, this will be [[None]]. */
+  protected[diplomacy] val parent: Option[LazyModule] = LazyModule.scope
 
-  // code snippets from 'InModuleBody' injection
+  /** Code snippets from [[InModuleBody]] injection. */
   protected[diplomacy] var inModuleBody = List[() => Unit]()
 
+  /** Chain of ancestor LazyModules, starting with [[parent]] on left. */
   def parents: Seq[LazyModule] = parent match {
     case None => Nil
     case Some(x) => x +: x.parents
   }
 
+  // [[LazyModule.scope]] stack push.
   LazyModule.scope = Some(this)
+  // Prepend this instance to [[parent]]'s list of children.
   parent.foreach(p => p.children = this :: p.children)
 
-  // suggestedName accumulates Some(names), taking the final one. Nones are ignored.
+  /** Accumulates Some(names), taking the final one. `None`s are ignored. */
   private var suggestedNameVar: Option[String] = None
+
+  /** Set instance name of [[LazyModuleImpLike]] module.
+    *
+    * @param x suggested name to [[LazyModuleImpLike]] module.
+    */
   def suggestName(x: String): this.type = suggestName(Some(x))
   def suggestName(x: Option[String]): this.type = {
     x.foreach { n => suggestedNameVar = Some(n) }
     this
   }
 
+  /** Helper function of [[className]]. This searches upwards in the class hierarchy for the first class name that was not an anonymous scala class.*/
   private def findClassName(c: Class[_]): String = {
     val n = c.getName.split('.').last
     if (n.contains('$')) findClassName(c.getSuperclass) else n
   }
 
-  lazy val className = findClassName(getClass)
-  lazy val suggestedName = suggestedNameVar.getOrElse(className)
-  lazy val desiredName = className // + hashcode?
+  /** The class name of this instance. */
+  lazy val className: String = findClassName(getClass)
+  /** Suggested instance name. */
+  lazy val suggestedName: String = suggestedNameVar.getOrElse(className)
+  /** Suggested module name. */
+  lazy val desiredName: String = className // + hashcode?
 
+  /** Return instance name. */
   def name = suggestedName // className + suggestedName ++ hashcode ?
+  /** Return source line that defines this instance. */
   def line = sourceLine(info)
 
   // Accessing these names can only be done after circuit elaboration!
-  lazy val moduleName = module.name // The final Verilog Module name
+  /** Module name, used in GraphML. */
```

Comment:
```suggestion
  /** Module name. Should only be accessed after circuit elaboration.*/
```

Reply:
IIIRC, `moduleName` is only used in `GraphML`?

Reply:
in the current rocket-chip codebase, yes, but nothing in this API prevents someone from using this function for some other purpose. I don't think "this function is used in XYZ other place" really makes sense as documentation. What do you think @richardxia ?

Reply:
I think you are right. used in GraphML sounds not like a documentation.